eng/pipelines/templates/steps/java-publishing.yml

```diff
@@ -59,18 +60,40 @@ steps:
           -InformationAction Continue
 
   - ${{if eq(parameters.Target, 'MavenCentral')}}:
-    - task: PowerShell@2
-      displayName: 'Publish to Maven Central'
-      inputs:
-        pwsh: true
-        workingDirectory: $(Agent.BuildDirectory)
-        filePath: ${{ parameters.JavaRepoRoot }}/eng/scripts/Publish-MavenPackages.ps1
-        arguments: >
-          -ArtifactDirectory ${{ parameters.ArtifactDirectory }}
-          -GroupIDFilter ${{ parameters.GroupID }}
-          -ArtifactIDFilter ${{ parameters.ArtifactID }}
-          -RepositoryUrl https://oss.sonatype.org/service/local/staging/deploy/maven2/
-          -RepositoryUsername azuresdk
-          -RepositoryPassword $(azuresdk-sonatype-password)
-          -GPGExecutablePath ${{ parameters.BuildToolsPath }}/tools/gpg/bin/gpg.exe
-          -InformationAction Continue
+
+    # Invoke this way if we are publishing a specific artifact (usually used from
+    # within universal pipelines).
+    - ${{if and(ne(parameters.GroupID, ''),ne(parameters.ArtifactID, ''))}}:
+      - task: PowerShell@2
+        displayName: 'Publish to Maven Central'
+        inputs:
+          pwsh: true
+          workingDirectory: $(Agent.BuildDirectory)
+          filePath: ${{ parameters.JavaRepoRoot }}/eng/scripts/Publish-MavenPackages.ps1
+          arguments: >
+            -ArtifactDirectory ${{ parameters.ArtifactDirectory }}
+            -GroupIDFilter ${{ parameters.GroupID }}
```

Comment:
As best I can tell no harm comes if you pass empty values for these into your script so why not just always call these? It also won't hurt to always pass StageOnly parameter. Unless I missed something I think all the other arguments are the same.